framework/wazuh/core/cluster/cluster.py

```diff
@@ -182,59 +179,50 @@ def walk_dir(dirname, recursive, files, excluded_files, excluded_extensions, get
-
-                # Create dict with metadata of 'full_path' file.
-                entry_metadata = {"mod_time": file_mod_time, 'cluster_item_key': get_cluster_item_key}
-                if '.merged' in entry:
-                    entry_metadata['merged'] = True
-                    entry_metadata['merge_type'] = 'agent-groups'
-                    entry_metadata['merge_name'] = os.path.join(dirname, entry)
-                else:
-                    entry_metadata['merged'] = False
-
-                if get_md5:
-                    entry_metadata['md5'] = md5(os.path.join(common.wazuh_path, full_path))
-
-                # Use the relative file path as a key to save its metadata dictionary.
-                walk_files[full_path] = entry_metadata
-
-            if recursive and path.isdir(os.path.join(common.wazuh_path, full_path)):
-                walk_files.update(walk_dir(full_path, recursive, files, excluded_files, excluded_extensions,
-                                           get_cluster_item_key, get_md5))
-
-        except Exception as e:
-            logger.debug(f"Could not get checksum of file {entry}: {e}")
-
+                    try:
+                        #  If 'all' files have been requested or entry is in the specified files list.
+                        if files == ['all'] or file_ in files:
+                            relative_file_path = path.join(path.relpath(root_, common.wazuh_path), file_)
+                            abs_file_path = path.join(root_, file_)
+                            file_mod_time = path.getmtime(abs_file_path)
+                            try:
+                                if file_mod_time == previous_status[relative_file_path]['mod_time']:
+                                    # The current file has not changed its mtime since the last integrity process.
+                                    walk_files[relative_file_path] = previous_status[relative_file_path]
+                                    continue
+                            except KeyError:
+                                pass
+                            # Create dict with metadata for the current file.
+                            file_metadata = {"mod_time": file_mod_time, 'cluster_item_key': get_cluster_item_key}
+                            if '.merged' in file_:
+                                file_metadata['merged'] = True
+                                file_metadata['merge_type'] = 'agent-groups'
+                                file_metadata['merge_name'] = abs_file_path
+                            else:
+                                file_metadata['merged'] = False
+                            if get_md5:
+                                file_metadata['md5'] = md5(abs_file_path)
+                            # Use the relative file path as a key to save its metadata dictionary.
+                            walk_files[relative_file_path] = file_metadata
+                    except Exception as e:
```

Comment:
We should add two different Exception catches for IOError. File not existing or any other problem with permissions.